### PR TITLE
[*] move models to ricos-types, fix selection d-e-s-t, use toolbars-v2 with detach commands

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/ricos-editor/web/src/content-modification/command-runner.spec.ts [hunk 1]
```diff
@@ -1,8 +1,6 @@
 import type { ParagraphNode } from 'ricos-content';
 import { Decoration_Type } from 'ricos-schema';
-import type { CommandDescriptor } from '../models/command';
-import type { RicosNodesRepository } from '../models/ricos-content';
-import type { DescriptorManager } from '../models/ricos-descriptor';
+import type { CommandDescriptor, RicosNodesRepository, DescriptorManager } from 'ricos-types';
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/content-modification/command-runner.spec.ts:3]: 👍

File: packages/ricos-editor/web/src/content-modification/command-runner.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
-import type { CommandDescriptor, CommandRunner, Commands } from '../models/command';
-import type { RicosNodesRepository } from '../models/ricos-content';
-import type { DescriptorManager } from '../models/ricos-descriptor';
+import type {
+  CommandDescriptor,
+  CommandRunner,
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/content-modification/command-runner.ts:3]: CommandRunner is internal as well

File: packages/ricos-editor/web/src/content-modification/commands/core-commands.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { setBold, unsetBold, toggleBold } from './bold';
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/content-modification/commands/core-commands.ts:1]: looks like `set`, `unset`, `toggle` could be refactored to

```ts
interface ToggleCommand<TDecoration> {
  set(paragraphs, decoration: TDecoration);
  unset(paragraphs, decoration: TDecoration);
  toggle(paragraphs, decoration: TDecoration);
}
```
and implemented once. Then any decoration-related command could use this logic.

File: packages/ricos-editor/web/src/content-modification/nodes/ricos-node-descriptor.ts [hunk 4]
```diff
@@ -1,6 +1,5 @@
-import type { RicosNode } from '../../models/ricos-content';
+import type { RicosNode, IRicosNodeDescriptor, RicosNodeMetadata } from 'ricos-types';
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/content-modification/nodes/ricos-node-descriptor.ts:1]: `IRicosNodeDescriptor`, `RicosNodeMetadata` are internal -- should not be exposed

File: packages/ricos-editor/web/src/content-modification/nodes/ricos-node-descriptors.ts [hunk 5]
```diff
@@ -1,6 +1,5 @@
-import type { RicosNode } from '../../models/ricos-content';
+import type { RicosNode, IRicosNodeDescriptors } from 'ricos-types';
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/content-modification/nodes/ricos-node-descriptors.ts:1]: `IRicosNodeDescriptors` is internal

File: packages/ricos-editor/web/src/content-modification/services/draft-content-repository-mock.ts [hunk 6]
```diff
@@ -1,8 +1,10 @@
 /* eslint-disable brace-style */
-import type { RicosNodesRepository } from '../../models/ricos-content';
-import type { IDraftEditorStateTranslator } from '../../models/draft-editor-state-translator';
+import type {
+  RicosNodesRepository,
+  IDraftEditorStateTranslator,
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/content-modification/services/draft-content-repository-mock.ts:4]: `IDraftEditorStateTranslator` is internal

File: packages/ricos-types/web/src/index.ts [hunk 7]
```diff
@@ -16,3 +16,4 @@ export * from './toolbarEnums';
 export * from './toolbarSettingsTypes';
 export * from './pluginsBiCallbacksTypes';
 export * from './editorCommandsType';
+export * from './models';
```
Comment (lxgreen) [inline packages/ricos-types/web/src/index.ts:19]: only public types that are necessary to define command are supposed to be exposed via `ricos-types`

File: packages/toolbars-v2/web/src/toolbarItemConfig/draftToolbarItemConfigDetachCommands.ts [hunk 8]
```diff
@@ -0,0 +1,271 @@
+// import { ContentResolver } from './ContentResolver';
```
Comment (lxgreen) [inline packages/toolbars-v2/web/src/toolbarItemConfig/draftToolbarItemConfigDetachCommands.ts:1]: where did this file come from?

**Review by lxgreen — CHANGES_REQUESTED** [review]
please see comments

File: packages/toolbars-v2/web/src/toolbarItemConfig/draftToolbarItemConfigDetachCommands.ts (hunk 8 above)
Comment (maimonav) [reply packages/toolbars-v2/web/src/toolbarItemConfig/draftToolbarItemConfigDetachCommands.ts:1]: That's where I've defined buttons that uses new commands, let's go over it today.

File: packages/ricos-editor/web/src/content-modification/commands/core-commands.ts (hunk 3 above)
Comment (maimonav) [reply packages/ricos-editor/web/src/content-modification/commands/core-commands.ts:1]: We can make create setDecoration, unsetDecoration & toggleDecoration, and can pass decoration to it,  let's discuss it today.

**Pull request merged**